core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreferWritePartitioning.java

```diff
@@ -46,6 +48,14 @@ public Result apply(TableWriterNode node, Captures captures, Context context)
             return Result.empty();
         }
 
+        double numberOfPartitions = getRowsCount(
+                context.getStatsProvider().getStats(node.getSource()),
+                node.getPreferredPartitioningScheme().get().getPartitioning().getColumns());
+
+        if (numberOfPartitions < getPreferredWritePartitioningThreshold(context.getSession())) {
```

Comment:
make this `<=`. This way we can always turn it on (even for single partition) even if stats are off.